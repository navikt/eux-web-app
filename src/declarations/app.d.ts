import { OptionTypeBase } from 'react-select'

export interface Option extends OptionTypeBase {
  label: string
  navn?: string
  value: string
}

export type Options = Array<Option>
export type Loading = {[key: string]: boolean}
export type Params = {[k: string] : string}
export interface ParamPayload {
  key: string,
  value?: any
}
