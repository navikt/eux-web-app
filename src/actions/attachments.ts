import * as types from 'constants/actionTypes'
import * as urls from 'constants/urls'
import {
  JoarkBrowserItem,
  JoarkBrowserItems,
  JoarkBrowserItemWithContent,
  JoarkList,
  JoarkPreview,
  SEDAttachmentPayloadWithFile
} from 'declarations/attachments.d'
import { ActionWithPayload, call } from '@navikt/fetch'
import mockJoark from 'mocks/attachments/joark'
import mockPreview from 'mocks/attachments/preview'
import { Action, ActionCreator } from 'redux'
const sprintf = require('sprintf-js').sprintf

export const createSavingAttachmentJob: ActionCreator<ActionWithPayload<JoarkBrowserItems>> = (
  joarkBrowserItems: JoarkBrowserItems
): ActionWithPayload<JoarkBrowserItems> => ({
  type: types.ATTACHMENT_SAVINGATTACHMENTJOB_SET,
  payload: joarkBrowserItems
})

export const getJoarkItemPreview = (
  item: JoarkBrowserItem
): ActionWithPayload<JoarkPreview> => {
  return call({
    url: sprintf(urls.API_JOARK_GET_URL, {
      dokumentInfoId: item.dokumentInfoId,
      journalpostId: item.journalpostId,
      variantformat: item.variant?.variantformat
    }),
    responseType: 'pdf',
    expectedPayload: mockPreview(),
    context: item,
    type: {
      request: types.ATTACHMENT_PREVIEW_REQUEST,
      success: types.ATTACHMENT_PREVIEW_SUCCESS,
      failure: types.ATTACHMENT_PREVIEW_FAILURE
    }
  })
}

export const listJoarkItems = (
  fnr: string, tema: string = ''
): ActionWithPayload<JoarkList> => {
  return call({
    url: sprintf(urls.API_ATTACHMENT_LIST_URL, { fnr, tema }),
    expectedPayload: mockJoark,
    type: {
      request: types.ATTACHMENT_LIST_REQUEST,
      success: types.ATTACHMENT_LIST_SUCCESS,
      failure: types.ATTACHMENT_LIST_FAILURE
    }
  })
}

export const resetSedAttachments: ActionCreator<Action> = (): Action => ({
  type: types.ATTACHMENT_RESET
})

export const sendAttachmentToSed = (
  params: SEDAttachmentPayloadWithFile, joarkBrowserItem: JoarkBrowserItem
): Action => {
  return call({
    url: sprintf(urls.API_JOARK_ATTACHMENT_URL, params),
    method: 'POST',
    cascadeFailureError: true,
    expectedPayload: joarkBrowserItem,
    context: {
      params,
      joarkBrowserItem
    },
    type: {
      request: types.ATTACHMENT_SEND_REQUEST,
      success: types.ATTACHMENT_SEND_SUCCESS,
      failure: types.ATTACHMENT_SEND_FAILURE
    }
  })
}

export const setJoarkItemPreview: ActionCreator<ActionWithPayload<JoarkBrowserItemWithContent | undefined>> = (
  item: JoarkBrowserItemWithContent | undefined
): ActionWithPayload<JoarkBrowserItemWithContent | undefined> => ({
  type: types.ATTACHMENT_PREVIEW_SET,
  payload: item
})
