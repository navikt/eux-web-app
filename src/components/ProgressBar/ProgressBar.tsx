import React from 'react'
import PT from 'prop-types'
import classNames from 'classnames'
import styled, { keyframes } from 'styled-components'

export type ProgressBarStatus = 'todo' | 'inprogress' | 'done' | 'error'

export interface ProgressBarProps {
  className?: string
  now?: number
  border ?: boolean
  status?: ProgressBarStatus
  children?: JSX.Element
}

const progressBarAnimation = keyframes`
  0% {background-position: 1rem 0;} to {background-position: 0 0;}
`

export const ProgressBarDiv = styled.div`
  background-color: var(--a-bg-default);
  width: 100%;
  display: block;
  border-radius: 4px;
  border-color: var(--a-border-default);
  border-width: 1px;
  &.border {
    border-style: solid;
  }
`

export const Bar = styled.div`
  border-radius: 4px;
  line-height: 2rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  white-space: nowrap;
  transition: width .3s ease;
  background-size: 1rem 1rem;
  animation: ${progressBarAnimation} 1s linear infinite;
  &.inprogress {
    background-color: var(--a-blue-500);
    color: var(--a-white);
    background-image: linear-gradient(45deg, hsla(0, 0%, 100%, .15) 25%, transparent 0, transparent 50%, hsla(0, 0%, 100%, .15) 0, hsla(0, 0%, 100%, .15) 75%, transparent 0, transparent);
  }
  &.done {
    background-color: var(--a-green-500);
    color: var(--a-white);
  }
  &.error {
    background-color: var(--a-red-500);
    color: var(--a-white);
  }
  &.todo {
     background-color: var(--a-border-divider);
  }
`

const ProgressBar: React.FC<ProgressBarProps> = ({
  className,
  now = 0,
  status = 'inprogress',
  children,
  border = true
}: ProgressBarProps): JSX.Element => (
  <ProgressBarDiv className={classNames(className, {border})}>
    <Bar
      role='progressbar'
      className={classNames(status)}
      style={{ width: (now) + '%' }}
      aria-valuenow={now}
      aria-valuemin={0}
      aria-valuemax={100}
    >
      {children}
    </Bar>
  </ProgressBarDiv>
)

ProgressBar.propTypes = {
  className: PT.string,
  now: PT.number.isRequired,
  status: PT.oneOf(['todo', 'inprogress', 'done', 'error'])
}

export default ProgressBar
