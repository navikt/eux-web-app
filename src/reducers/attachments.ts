import * as types from 'constants/actionTypes'
import {
  SavingAttachmentsJob,
  JoarkPoster,
  JoarkBrowserItem, JoarkBrowserItems
} from 'declarations/attachments'
import { ActionWithPayload } from '@navikt/fetch'
import _ from 'lodash'
import { AnyAction } from 'redux'

export interface JoarkState {
  list: Array<JoarkPoster> | undefined
  previewFileRaw: Blob | null | undefined
  previewAttachmentFileRaw: Blob | null | undefined
  savingAttachmentsJob: SavingAttachmentsJob | undefined
}

export const initialJoarkState: JoarkState = {
  list: undefined,
  previewFileRaw: undefined,
  previewAttachmentFileRaw: undefined,
  savingAttachmentsJob: undefined
}

const joarkReducer = (state: JoarkState = initialJoarkState, action: AnyAction): JoarkState => {
  switch (action.type) {
    case types.APP_RESET:
      return initialJoarkState

    case types.ATTACHMENT_LIST_SUCCESS:
      return {
        ...state,
        list: action.payload.data.dokumentoversiktBruker.journalposter
      }

    case types.ATTACHMENT_PREVIEW_SET:
      return {
        ...state,
        previewFileRaw: action.payload
      }

    case types.ATTACHMENT_PREVIEW_REQUEST:
      return {
        ...state,
        previewFileRaw: undefined
      }

    case types.ATTACHMENT_PREVIEW_FAILURE:
      return {
        ...state,
        previewFileRaw: null
      }

    case types.ATTACHMENT_PREVIEW_SUCCESS:
      return {
        ...state,
        previewFileRaw: action.payload
      }

    case types.ATTACHMENT_FROM_RINA_PREVIEW_SET:
      return {
        ...state,
        previewAttachmentFileRaw: action.payload
      }

    case types.ATTACHMENT_FROM_RINA_PREVIEW_REQUEST:
      return {
        ...state,
        previewAttachmentFileRaw: undefined
      }

    case types.ATTACHMENT_FROM_RINA_PREVIEW_FAILURE:
      return {
        ...state,
        previewAttachmentFileRaw: null
      }

    case types.ATTACHMENT_FROM_RINA_PREVIEW_SUCCESS:
      return {
        ...state,
        previewAttachmentFileRaw: action.payload
      }



    case types.ATTACHMENT_SEND_REQUEST:
      return {
        ...state,
        savingAttachmentsJob: {
          ...(state.savingAttachmentsJob as SavingAttachmentsJob),
          saving: action.context.joarkBrowserItem
        }
      }

    case types.ATTACHMENT_SEND_SUCCESS: {
      const newlySavedJoarkBrowserItem: JoarkBrowserItem = (action as ActionWithPayload).context.joarkBrowserItem
      const newRemaining = _.reject(state.savingAttachmentsJob!.remaining, (item: JoarkBrowserItem) => {
        return item.dokumentInfoId === newlySavedJoarkBrowserItem.dokumentInfoId &&
          item.journalpostId === newlySavedJoarkBrowserItem.journalpostId &&
          item.variant === newlySavedJoarkBrowserItem.variant
      })
      newlySavedJoarkBrowserItem.type = 'sednew'

      const newSaved: JoarkBrowserItems = state.savingAttachmentsJob!.saved.concat(newlySavedJoarkBrowserItem)

      return {
        ...state,
        savingAttachmentsJob: {
          total: state.savingAttachmentsJob!.total,
          saving: undefined,
          saved: newSaved,
          remaining: newRemaining
        }
      }
    }

    case types.ATTACHMENT_SEND_FAILURE: {
      return {
        ...state,
        savingAttachmentsJob: undefined
      }
    }

    case types.ATTACHMENT_SAVINGATTACHMENTJOB_SET:

      return {
        ...state,
        savingAttachmentsJob: {
          total: (action as ActionWithPayload).payload,
          remaining: (action as ActionWithPayload).payload,
          saving: undefined,
          saved: []
        }
      }

    case types.ATTACHMENT_RESET: {
      return {
        ...state,
        savingAttachmentsJob: undefined
      }
    }

    default:
      return state
  }
}

export default joarkReducer
