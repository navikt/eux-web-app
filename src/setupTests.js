/* eslint-disable */

import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import each from 'jest-each';

global.fetch = require('jest-fetch-mock');

Enzyme.configure({ adapter: new Adapter() });

global.shallow = shallow;
global.mount = mount;

global.each = each;
