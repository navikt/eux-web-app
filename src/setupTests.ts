import {render, screen} from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import { useDispatch, useSelector } from 'react-redux'
import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'

jest.mock('react-redux');

(global as any).screen = screen;
(global as any).render = render;
(global as any).act = act

// eslint-disable-next-line no-undef
HTMLCanvasElement.prototype.getContext = jest.fn()
window.scrollTo = jest.fn()

Object.defineProperty(window, 'getComputedStyle', {
  value: () => ({
    getPropertyValue: () => ('')
  })
})

jest.mock('amplitude-js', () => ({
  getInstance: () => ({
    init: jest.fn(),
    logEvent: jest.fn()
  })
}))

jest.mock('i18next', () => {
  const use = jest.fn()
  const init = jest.fn()
  const loadLanguages = jest.fn()
  const result = {
    use: use,
    init: init,
    loadLanguages: loadLanguages
  }
  use.mockImplementation(() => result)
  return result
})

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

export const stageSelector = (defaultSelector: any, params: any) => {
  (useDispatch as jest.Mock).mockImplementation(() => jest.fn());
  (useSelector as jest.Mock).mockImplementation(() => ({
    ...defaultSelector,
    ...params
  }))
}
