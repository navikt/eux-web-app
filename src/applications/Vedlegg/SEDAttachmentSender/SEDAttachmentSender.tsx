import { IS_TEST } from 'constants/environment'
import {
  JoarkBrowserItem,
  SavingAttachmentsJob,
  SEDAttachmentPayload,
  SEDAttachmentPayloadWithFile
} from 'declarations/attachments.d'
import {
  SEDAttachmentPayloadPropType
} from 'declarations/attachments.pt'
import { State } from 'declarations/reducers'
import ProgressBar, { ProgressBarStatus } from '@navikt/fremdriftslinje'
import _ from 'lodash'
import md5 from 'md5'
import { Button } from '@navikt/ds-react'
import { HorizontalSeparatorDiv } from '@navikt/hoykontrast'
import PT from 'prop-types'
import React, { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useSelector } from 'react-redux'
import styled from 'styled-components'

const SEDAttachmentSenderDiv = styled.div`
  display: flex;
  align-items: center;
`

export interface SEDAttachmentSenderProps {
  attachmentsError ?: boolean
  className?: string
  initialStatus ?: ProgressBarStatus
  onCancel ?: () => void
  onFinished : (summary: any) => void
  onSaved: (savingAttachmentsJob: SavingAttachmentsJob) => void
  payload: SEDAttachmentPayload
  sendAttachmentToSed : (params: SEDAttachmentPayloadWithFile, unsent: JoarkBrowserItem) => void
}

export interface SEDAttachmentSelector {
  savingAttachmentsJob: SavingAttachmentsJob | undefined
}

const mapState = (state: State): SEDAttachmentSelector => ({
  savingAttachmentsJob: state.attachments.savingAttachmentsJob
})

const SEDAttachmentSender: React.FC<SEDAttachmentSenderProps> = ({
  attachmentsError, className, initialStatus = 'inprogress',
  onCancel, onFinished, onSaved, payload, sendAttachmentToSed
}: SEDAttachmentSenderProps): JSX.Element => {
  const [_status, setStatus] = useState<ProgressBarStatus>(initialStatus)
  const { savingAttachmentsJob }: SEDAttachmentSelector = useSelector<State, SEDAttachmentSelector>(mapState)
  const { t } = useTranslation()

  useEffect(() => {
    if (savingAttachmentsJob) {
      // all attachments are sent - conclude
      if (savingAttachmentsJob.remaining.length === 0) {
        if (!savingAttachmentsJob.saving) {
          /* istanbul ignore next */
          const summary = {
            total: savingAttachmentsJob.total.length,
            saved: savingAttachmentsJob.saved.length
          }
          if (!IS_TEST) {
            console.log('Concluding.' +
              ' Total (' + savingAttachmentsJob.total.length +
              ') saved (' + savingAttachmentsJob.saved.length + ') ' +
              ' saving (' + !!savingAttachmentsJob.saving +
              ') remaining (' + savingAttachmentsJob.remaining.length + ')')
          }
          onSaved(savingAttachmentsJob)
          setStatus('done')
          onFinished(summary)
          return
        }
        return
      // still are attachments to send,
      }

      // one attachment was just saved. Pick another one to save
      if (!savingAttachmentsJob.saving) {
        /* istanbul ignore next */
        if (!IS_TEST) {
          console.log('Picking one to save.' +
            ' Total (' + savingAttachmentsJob.total.length +
            ') saved (' + savingAttachmentsJob.saved.length + ') ' +
            ' saving (' + !!savingAttachmentsJob.saving +
            ') remaining (' + savingAttachmentsJob.remaining.length + ')')
        }
        onSaved(savingAttachmentsJob)
        const unsentAttachment: JoarkBrowserItem = _.first(savingAttachmentsJob.remaining)!
        const params: SEDAttachmentPayloadWithFile = {
          ...payload,
          journalpostId: unsentAttachment.journalpostId,
          dokumentInfoId: unsentAttachment.dokumentInfoId!,
          variantformat: unsentAttachment.variant!.variantformat!,
          filnavn: unsentAttachment.title.replaceAll(' ', '_') ?? 'fil-' + md5('' + new Date().getTime())
        }
        sendAttachmentToSed(params, unsentAttachment)
      } else {
        // one attachment will be saved now. Display as such.
        /* istanbul ignore next */
        if (!IS_TEST) {
          console.log('Saving.' +
            ' Total (' + savingAttachmentsJob.total.length +
            ') saved (' + savingAttachmentsJob.saved.length + ') ' +
            ' saving (' + !!savingAttachmentsJob.saving +
            ') remaining (' + savingAttachmentsJob.remaining.length + ')')
        }
      }
    }
  }, [onSaved, onFinished, payload, sendAttachmentToSed, savingAttachmentsJob])

  useEffect(() => {
    if (attachmentsError && _status !== 'error') {
      setStatus('error')
    }
  }, [attachmentsError, setStatus, _status])

  if (!savingAttachmentsJob) {
    return <div />
  }

  const current: number = (savingAttachmentsJob.saved.length + (savingAttachmentsJob.saving ? 1 : 0))
  const total: number = savingAttachmentsJob.total.length
  const percentage: number = (Math.floor((current * 100) / total))

  return (
    <SEDAttachmentSenderDiv
      data-testid='c-sedAttachmentSender__div-id'
      className={className}
    >
      <ProgressBar
        data-testid='c-sedAttachmentSender__progress-bar-id'
        now={percentage}
        status={_status}
      >
        <>
          {_status === 'inprogress' && t('message:loading-sending-XofY', {
            current: current,
            total: total
          })}
          {_status === 'done' && t('label:vedlegg-sendt')}
          {_status === 'error' && t('message:error-sending-attachments')}
        </>
      </ProgressBar>
      {_status === 'inprogress' && _.isFunction(onCancel) && (
        <>
          <HorizontalSeparatorDiv data-sise='0.35' />
          <Button
            variant='secondary'
            data-testid='c-sedAttachmentSender__cancel-button-id'
            onClick={(e: React.MouseEvent<HTMLButtonElement>) => {
              e.preventDefault()
              e.stopPropagation()
              onCancel()
            }}
          >
            {t('el:button-cancel')}
          </Button>
        </>
      )}
    </SEDAttachmentSenderDiv>
  )
}

SEDAttachmentSender.propTypes = {
  attachmentsError: PT.bool,
  className: PT.string,
  initialStatus: PT.oneOf(['todo', 'inprogress', 'done', 'error']),
  onCancel: PT.func,
  onFinished: PT.func.isRequired,
  onSaved: PT.func.isRequired,
  payload: SEDAttachmentPayloadPropType.isRequired,
  sendAttachmentToSed: PT.func.isRequired
}

export default SEDAttachmentSender
