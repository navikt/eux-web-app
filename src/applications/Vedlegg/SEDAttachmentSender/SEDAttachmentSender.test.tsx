import { SavingAttachmentsJob } from 'declarations/attachments'
import { mount, ReactWrapper } from 'enzyme'
import ProgressBar from '@navikt/fremdriftslinje'
import React from 'react'
import { stageSelector } from 'setupTests'
import SEDAttachmentSender, { SEDAttachmentSenderProps } from './SEDAttachmentSender'
import joarkBrowserItems from 'mocks/attachments/items'

const defaultSelector = {
  savingAttachmentsJob: {
    total: joarkBrowserItems,
    saved: [],
    saving: undefined,
    remaining: joarkBrowserItems
  }
}

describe('components/SEDAttachmentSender/SEDAttachmentSender', () => {
  let wrapper: ReactWrapper
  const initialMockProps: SEDAttachmentSenderProps = {
    attachmentsError: false,
    className: 'mock-sedAttachmentSender',
    initialStatus: 'inprogress',
    onCancel: jest.fn(),
    onFinished: jest.fn(),
    onSaved: jest.fn(),
    payload: {
      rinaId: '456',
      rinaDokumentId: '789'
    },
    sendAttachmentToSed: jest.fn()
  }

  beforeEach(() => {
    stageSelector(defaultSelector, {})
    wrapper = mount(<SEDAttachmentSender {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Render: match snapshot', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Render: Has proper HTML structure', () => {
    expect(wrapper.exists('[data-test-id=\'c-sedAttachmentSender__div-id\']')).toBeTruthy()
    expect(wrapper.exists('[data-test-id=\'c-sedAttachmentSender__progress-bar-id\']')).toBeTruthy()
    expect(wrapper.find(ProgressBar).props().status).toEqual(initialMockProps.initialStatus)
  })

  it('Handling: cancel button pressed', () => {
    (initialMockProps.onCancel as jest.Mock).mockReset()
    wrapper.find('[data-test-id=\'c-sedAttachmentSender__cancel-button-id\']').hostNodes().simulate('click')
    expect(initialMockProps.onCancel).toHaveBeenCalled()
  })

  it('Handling: finished when no more items', () => {
    const mockSavingAttachmentJob: SavingAttachmentsJob = {
      total: joarkBrowserItems,
      saved: joarkBrowserItems,
      saving: undefined,
      remaining: []
    }
    stageSelector(defaultSelector, {
      savingAttachmentsJob: mockSavingAttachmentJob
    })
    wrapper = mount(<SEDAttachmentSender {...initialMockProps} />)
    expect(wrapper.find(ProgressBar).props().status).toEqual('done')
    expect(initialMockProps.onSaved).toHaveBeenCalledWith(mockSavingAttachmentJob)
    expect(initialMockProps.onFinished).toHaveBeenCalled()
  })
})
