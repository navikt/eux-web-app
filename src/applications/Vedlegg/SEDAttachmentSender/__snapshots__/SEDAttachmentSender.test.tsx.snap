// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/BUC/components/SEDAttachmentSender/SEDAttachmentSender Render: match snapshot 1`] = `
.c5 {
  background-color: inherit;
}

ui-fss.g13[id="c4"] {
  content: "c5,";
}

.c3 {
  display: inline-block;
  margin-left: 1rem;
}

ui-fss.g23[id="c2"] {
  content: "c3,";
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

ui-fss.g28[id="c0"] {
  content: "c1,";
}

<SEDAttachmentSender
  attachmentsError={false}
  className="mock-sedAttachmentSender"
  initialStatus="inprogress"
  onCancel={[MockFunction]}
  onFinished={[MockFunction]}
  onSaved={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "remaining": Array [
              Object {
                "date": 1970-01-31T23:00:00.000Z,
                "dokumentInfoId": "4",
                "hasSubrows": false,
                "journalpostId": "1",
                "key": "1",
                "tema": "tema",
                "title": "title",
                "type": "joark",
                "variant": Object {
                  "filnavn": "mockFilnavn",
                  "variantformat": "mockVariant",
                },
              },
            ],
            "saved": Array [],
            "saving": undefined,
            "total": Array [
              Object {
                "date": 1970-01-31T23:00:00.000Z,
                "dokumentInfoId": "4",
                "hasSubrows": false,
                "journalpostId": "1",
                "key": "1",
                "tema": "tema",
                "title": "title",
                "type": "joark",
                "variant": Object {
                  "filnavn": "mockFilnavn",
                  "variantformat": "mockVariant",
                },
              },
            ],
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  payload={
    Object {
      "aktoerId": "123",
      "rinaDokumentId": "789",
      "rinaId": "456",
    }
  }
  sendAttachmentToSed={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "aktoerId": "123",
            "dokumentInfoId": "4",
            "journalpostId": "1",
            "rinaDokumentId": "789",
            "rinaId": "456",
            "variantformat": "mockVariant",
          },
          Object {
            "date": 1970-01-31T23:00:00.000Z,
            "dokumentInfoId": "4",
            "hasSubrows": false,
            "journalpostId": "1",
            "key": "1",
            "tema": "tema",
            "title": "title",
            "type": "joark",
            "variant": Object {
              "filnavn": "mockFilnavn",
              "variantformat": "mockVariant",
            },
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <styled.div
    className="mock-sedAttachmentSender"
    data-testid="a-buc-c-sedAttachmentSender__div-id"
  >
    <div
      className="c0 c1 mock-sedAttachmentSender"
      data-testid="a-buc-c-sedAttachmentSender__div-id"
    >
      <h
        data-testid="a-buc-c-sedAttachmentSender__progress-bar-id"
        now={0}
        status="inprogress"
      >
        <Ze
          highContrast={false}
        >
          <De
            theme={
              Object {
                "alternative-background-color": "#e9e7e7",
                "black": "#000000",
                "border-radius-base": "4px",
                "border-radius-large": "6px",
                "border-radius-small": "2px",
                "container-desktop": "(940px + @grid-gutter-width)",
                "container-large-desktop": "(1140px + @grid-gutter-width)",
                "container-lg": "(1140px + @grid-gutter-width)",
                "container-md": "(940px + @grid-gutter-width)",
                "container-sm": "(720px + @grid-gutter-width)",
                "container-tablet": "(720px + @grid-gutter-width)",
                "fokusFarge": "#01b19d",
                "grayBackground": "#e9e7e7",
                "grayIcon": "#b7b1a9",
                "grayInactive": "#78706a",
                "grayModia": "#333333",
                "grid-columns": "12",
                "grid-float-breakpoint": "768px",
                "grid-float-breakpoint-max": "(@grid-float-breakpoint - 1)",
                "grid-gutter-width": "16px",
                "inverted-font-color": "#FFFFFF",
                "line-height-large": "1.3333333",
                "line-height-small": "1.5",
                "main-active-color": "#43dfNaN",
                "main-background-color": "#FFFFFF",
                "main-border-color": "#78706a",
                "main-border-radius": "4px",
                "main-disabled-color": "#b7b1a9",
                "main-error-color": "#ba3a26",
                "main-focus-color": "#01b19d",
                "main-font-color": "#3e3832",
                "main-hover-color": "#20f4NaN",
                "main-interactive-color": "#0067c5",
                "main-success-color": "#06893a",
                "main-warning-color": "#ff9100",
                "navBakgrunn": "#ffffff",
                "navBla": "#0067c5",
                "navBlaDarken20": "#007fb7",
                "navBlaDarken40": "#0198aa",
                "navBlaDarken60": "#01b19d",
                "navBlaDarken80": "#02ca90",
                "navBlaLighten20": "#0385NaN",
                "navBlaLighten40": "#06a3NaN",
                "navBlaLighten60": "#09c2NaN",
                "navBlaLighten80": "#0ce0NaN",
                "navDypBla": "#005b82",
                "navDypBlaDarken20": "#007682",
                "navDypBlaDarken40": "#019182",
                "navDypBlaDarken60": "#01ac82",
                "navDypBlaDarken80": "#02c782",
                "navDypBlaLighten20": "#037bNaN",
                "navDypBlaLighten40": "#069cNaN",
                "navDypBlaLighten60": "#09bdNaN",
                "navDypBlaLighten80": "#0cdeNaN",
                "navGra20": "#c6c2bf",
                "navGra40": "#b7b1a9",
                "navGra60": "#78706a",
                "navGra80": "#59514b",
                "navGraBakgrunn": "#f1f1f1",
                "navGronn": "#06893a",
                "navGronnDarken20": "#059b48",
                "navGronnDarken40": "#04ad57",
                "navGronnDarken60": "#04bf65",
                "navGronnDarken80": "#03d174",
                "navGronnLighten20": "#07a0NaN",
                "navGronnLighten40": "#09b8NaN",
                "navGronnLighten60": "#0bcfNaN",
                "navGronnLighten80": "#0de7NaN",
                "navLilla": "#634689",
                "navLillaDarken20": "#4f6587",
                "navLillaDarken40": "#3c8486",
                "navLillaDarken60": "#29a485",
                "navLillaDarken80": "#16c384",
                "navLillaLighten20": "#526bNaN",
                "navLillaLighten40": "#4190NaN",
                "navLillaLighten60": "#30b5NaN",
                "navLillaLighten80": "#1fdaNaN",
                "navLimeGronn": "#a2ad00",
                "navLimeGronnDarken20": "#82b71a",
                "navLimeGronnDarken40": "#62c234",
                "navLimeGronnDarken60": "#42cd4e",
                "navLimeGronnDarken80": "#22d868",
                "navLimeGronnLighten20": "#84bdNaN",
                "navLimeGronnLighten40": "#67cdNaN",
                "navLimeGronnLighten60": "#49deNaN",
                "navLimeGronnLighten80": "#2ceeNaN",
                "navLysBla": "#66cbec",
                "navLysBlaDarken20": "#52cfd7",
                "navLysBlaDarken40": "#3ed4c2",
                "navLysBlaDarken60": "#2ad9ad",
                "navLysBlaDarken80": "#16de98",
                "navLysBlaLighten20": "#54d5NaN",
                "navLysBlaLighten40": "#43dfNaN",
                "navLysBlaLighten60": "#31eaNaN",
                "navLysBlaLighten80": "#20f4NaN",
                "navLysGra": "#e9e7e7",
                "navMorkGra": "#3e3832",
                "navOransje": "#ff9100",
                "navOransjeDarken20": "#cca11a",
                "navOransjeDarken40": "#9ab134",
                "navOransjeDarken60": "#67c24e",
                "navOransjeDarken80": "#35d268",
                "navOransjeLighten20": "#cfa7NaN",
                "navOransjeLighten40": "#9fbdNaN",
                "navOransjeLighten60": "#6fd3NaN",
                "navOransjeLighten80": "#3fe9NaN",
                "navRod": "#c30000",
                "navRodDarken20": "#9c2d1a",
                "navRodDarken40": "#765a34",
                "navRodDarken60": "#4f884e",
                "navRodDarken80": "#29b568",
                "navRodLighten20": "#9f33NaN",
                "navRodLighten40": "#7b66NaN",
                "navRodLighten60": "#5799NaN",
                "navRodLighten80": "#33ccNaN",
                "orangeFocus": "#ffbd66",
                "orangeFocusDarken20": "#ccc46b",
                "orangeFocusDarken40": "#9acc71",
                "orangeFocusDarken60": "#67d377",
                "orangeFocusDarken80": "#35db7d",
                "orangeFocusLighten20": "#cfcaNaN",
                "orangeFocusLighten40": "#9fd7NaN",
                "orangeFocusLighten60": "#6fe4NaN",
                "orangeFocusLighten80": "#3ff1NaN",
                "padding-base-horizontal": "12px",
                "padding-base-vertical": "6px",
                "padding-large-horizontal": "16px",
                "padding-large-vertical": "10px",
                "padding-small-horizontal": "10px",
                "padding-small-vertical": "5px",
                "padding-xs-horizontal": "5px",
                "padding-xs-vertical": "1px",
                "pinkErrorBg": "#f3e3e3",
                "redError": "#ba3a26",
                "redErrorDarken20": "#955b38",
                "redErrorDarken40": "#707d4b",
                "redErrorDarken60": "#4c9f5d",
                "redErrorDarken80": "#27c170",
                "redErrorLighten20": "#9761NaN",
                "redErrorLighten40": "#7588NaN",
                "redErrorLighten60": "#53b0NaN",
                "redErrorLighten80": "#31d7NaN",
                "screen-desktop": "992px",
                "screen-lg": "1200px",
                "screen-lg-desktop": "1200px",
                "screen-lg-min": "1200px",
                "screen-md": "992px",
                "screen-md-max": "(@screen-lg-min - 1)",
                "screen-md-min": "992px",
                "screen-sm": "768px",
                "screen-sm-max": "(@screen-md-min - 1)",
                "screen-sm-min": "768px",
                "screen-tablet": "768px",
                "screen-xs-max": "(@screen-sm-min - 1)",
                "type": "theme",
                "white": "#FFFFFF",
              }
            }
          >
            <styled.div
              className=""
            >
              <div
                className="c0 oCzoH"
              >
                <styled.div
                  aria-valuemax={100}
                  aria-valuemin={0}
                  aria-valuenow={0}
                  className="inprogress"
                  role="progressbar"
                  style={
                    Object {
                      "width": "0%",
                    }
                  }
                >
                  <div
                    aria-valuemax={100}
                    aria-valuemin={0}
                    aria-valuenow={0}
                    className="klNAus inprogress"
                    role="progressbar"
                    style={
                      Object {
                        "width": "0%",
                      }
                    }
                  >
                    buc:loading-sendingXofY
                  </div>
                </styled.div>
              </div>
            </styled.div>
          </De>
        </Ze>
      </h>
      <styled.div
        data-sise="0.35"
      >
        <div
          className="c2 c3"
          data-sise="0.35"
        />
      </styled.div>
      <Styled(o)
        data-testid="a-buc-c-sedAttachmentSender__cancel-button-id"
        kompakt={true}
        mini={true}
        onClick={[Function]}
      >
        <o
          className="c4 c5"
          data-testid="a-buc-c-sedAttachmentSender__cancel-button-id"
          kompakt={true}
          mini={true}
          onClick={[Function]}
        >
          <c
            autoDisableVedSpinner={false}
            className="c4 c5"
            data-testid="a-buc-c-sedAttachmentSender__cancel-button-id"
            disabled={false}
            htmlType="submit"
            inaktivKlasseVedDisabled={false}
            kompakt={true}
            mini={true}
            onClick={[Function]}
            spinner={false}
            type="standard"
          >
            <button
              className="knapp c4 c5 knapp--standard knapp--mini knapp--kompakt"
              data-testid="a-buc-c-sedAttachmentSender__cancel-button-id"
              disabled={false}
              onClick={[Function]}
              type="submit"
            >
              ui:cancel
            </button>
          </c>
        </o>
      </Styled(o)>
    </div>
  </styled.div>
</SEDAttachmentSender>
`;
